### PR TITLE
moved ref system to api

### PR DESCRIPTION
**Pull request by matthias-luger**

### CODE REVIEW

**Comment by matthias-luger** [comment]
Need to be tested

File: api/ApiHelper.tsx [hunk 1]
```diff
@@ -926,9 +925,19 @@ export function initAPI(returnSSRResponse: boolean = false): API {
...
+            }
+
+            httpApi.sendApiRequest({
                 type: RequestType.GET_REF_INFO,
                 data: '',
+                requestHeader: {
+                    GoogleToken: googleId
+                },
                 resolve: (response: any) => {
                     resolve(parseRefInfo(response))
```
Comment (Ekwav) [inline api/ApiHelper.tsx:942]: the format changed

File: api/ApiHelper.tsx [hunk 2]
```diff
@@ -926,9 +925,19 @@ export function initAPI(returnSSRResponse: boolean = false): API {
 
     let getRefInfo = (): Promise<RefInfo> => {
         return new Promise((resolve, reject) => {
-            websocketHelper.sendRequest({
+            let googleId = localStorage.getItem('googleId')
+            if (!googleId) {
+                toast.error('You need to be logged in to use the ref system something.')
```
Comment (Ekwav) [inline api/ApiHelper.tsx]:
```suggestion
                toast.error('You need to be logged in to use the ref system.')
```

File: api/ApiHelper.tsx [hunk 3]
```diff
@@ -942,17 +951,33 @@ export function initAPI(returnSSRResponse: boolean = false): API {
...
     let setRef = (refId: string): Promise<void> => {
         return new Promise((resolve, reject) => {
-            websocketHelper.sendRequest({
-                type: RequestType.SET_REF,
-                data: refId,
-                resolve: () => {
-                    resolve()
+            let googleId = localStorage.getItem('googleId')
+            if (!googleId) {
+                toast.error('You need to be logged in to use the ref system something.')
```
Comment (Ekwav) [inline api/ApiHelper.tsx]: I don't think showing an error to new users is a good idea.

**Review by Ekwav — CHANGES_REQUESTED** [review]
Neither requests work, setRef isn't even called according to the network tab

File: api/ApiHelper.tsx (hunk 3 above)
Comment (matthias-luger) [reply api/ApiHelper.tsx]: This should really never occur as the function is called in the onAfterLogin.
But if for some reason it is called before, why should I not send out an error? What should I do instead?

**Review by Ekwav — APPROVED** [review]

**Pull request merged**